examples/web-wagmi/src/upload.ts

```diff
@@ -1,41 +1,78 @@
+import { Readable } from 'stream';
+
 import { WebBundlr } from '@bundlr-network/client';
+import { ReadableWebToNodeStream } from 'readable-web-to-node-stream';
 import { fetchSigner } from 'wagmi/actions';
 
+import { ILocalFile, ImageType } from './hooks/useFileSelect';
 import { never } from './utils';
 
 const TOP_UP = '200000000000000000'; // 0.2 MATIC
 const MIN_FUNDS = 0.05;
 
+async function getBundlr() {
+  const signer = await fetchSigner();
+
+  const bundlr = new WebBundlr('https://devnet.bundlr.network', 'matic', signer?.provider, {
+    providerUrl: 'https://rpc-mumbai.maticvigil.com/',
+  });
+
+  await bundlr.ready();
+
+  const balance = await bundlr.getBalance((await signer?.getAddress()) ?? never());
+
+  if (bundlr.utils.unitConverter(balance).toNumber() < MIN_FUNDS) {
+    await bundlr.fund(TOP_UP);
+  }
+
+  return bundlr;
+}
+
 export async function upload(data: unknown): Promise<string> {
   const confirm = window.confirm(
     `We will now upload metadata file via the Bundlr Network.
 
 Please make sure your wallet is connected to the Polygon Mumbai testnet.
 
-You can get some Mumbai MATIC from the Mumbai Faucet: https://mumbaifaucet.com/`,
+You can get some Mumbai MATIC from the Mumbai Faucet: 
+https://mumbaifaucet.com/ or https://faucet.polygon.technology/`,
   );
 
   if (!confirm) {
     throw new Error('User cancelled');
   }
 
-  const signer = await fetchSigner();
+  const bundlr = await getBundlr();
 
-  const bundlr = new WebBundlr('https://devnet.bundlr.network', 'matic', signer?.provider, {
-    providerUrl: 'https://rpc-mumbai.maticvigil.com/',
+  const serialized = JSON.stringify(data);
+  const tx = await bundlr.upload(serialized, {
+    tags: [{ name: 'Content-Type', value: 'application/json' }],
   });
 
-  await bundlr.ready();
+  return `https://arweave.net/${tx.id}`;
+}
 
-  const balance = await bundlr.getBalance((await signer?.getAddress()) ?? never());
+export async function uploadImage(file: ILocalFile<ImageType>): Promise<string> {
+  const confirm = window.confirm(
+    `We will now upload an image via the Bundlr Network to Arweave.
 
-  if (bundlr.utils.unitConverter(balance).toNumber() < MIN_FUNDS) {
-    await bundlr.fund(TOP_UP);
+Please make sure your wallet is connected to the Polygon Mumbai testnet.
+
+You can get some Mumbai MATIC from the Mumbai Faucet:
+https://mumbaifaucet.com/ or https://faucet.polygon.technology/`,
+  );
+
+  if (!confirm) {
+    throw new Error('User cancelled');
   }
 
-  const serialized = JSON.stringify(data);
-  const tx = await bundlr.upload(serialized, {
-    tags: [{ name: 'Content-Type', value: 'application/json' }],
+  const bundlr = await getBundlr();
+
+  const webStream = file.stream();
+  const nodeStream = new ReadableWebToNodeStream(webStream);
```

Comment:
just curious, any specific reason we are not using the guide from bundler docs? https://docs.bundlr.network/tutorials/file-uploader
I see they are using a package called `filereader-stream`.

Reply:
first of all `filereader-stream` doesn't provide typescript types, it's old and depends on even older modules. I didn't want to include it in our example.
I've spent a few hours trying to understand what it is doing and simplify it for our use case. Unfortunately `WebBundlr` implementation is heavily based on node/common version and works with node.js types, so the problem was how to make it work with web stream or buffer.
Cesare helped me here with finding ReadableWebToNodeStream module that glued it all together.
In general `WebBundlr` is not really web-friendly 😄